File: 염성훈/10.Greedy_03/pro_큰 수 만들기.py
def solution(number, k):
    stack = [number[0]]

    for num in number[1:]:
        # num의 값이 stack의 가장 상단의 값보다 크면, 기존의 값을 제거하고 새로운 값으로 대치한다.
        while len(stack) > 0 and stack[-1] < num and k > 0:
            k -= 1
            stack.pop()
        stack.append(num)

    if k!= 0:
        stack = stack[:-k]

    return ''.join(stack)

number = "4177252841"
k = 4

print(solution(number,k))

File: AHYEON/10.GREEDY/pro_큰수만들기.swift
import Foundation

func solution(_ number:String, _ k:Int) -> String {
    var result = ""
    var stack: String = ""
    var k = k

    number.forEach { num in
        while true {
            guard let peek = stack.last, Int(String(peek))! < Int(String(num))! else { break }
            if k == 0 { break }
            k -= 1
            stack.removeLast()
        }
        stack.append(num)
    }

    result = stack

    if k > 0  {
        result = String(stack.dropLast(k))
    }

    return result
}



